android-agent/src/test/java/io/opentelemetry/android/instrumentation/AndroidInstrumentationRegistryImplTest.kt

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.android.instrumentation
+
+import android.app.Application
+import io.mockk.mockk
+import io.opentelemetry.android.OpenTelemetryRum
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.fail
+
+class AndroidInstrumentationRegistryImplTest {
+    private lateinit var registry: AndroidInstrumentationRegistryImpl
+
+    @BeforeEach
+    fun setUp() {
+        registry = AndroidInstrumentationRegistryImpl()
+    }
+
+    @Test
+    fun `Find and register implementations available in the classpath when querying an instrumentation`() {
+        val instrumentation = registry.get(TestAndroidInstrumentation::class.java)
+
+        instrumentation?.install(mockk(), mockk())
```

Comment:
Should we also assert that `installed` is false before we call install? Just to kinda assert that the registry itself doesn't do any installation.

Reply:
Good point, I've just updated it.